### ISSUE TITLE
SQL error upon enabling encryption

### ISSUE THREAD

**Issue by joesfer** [laurent22/joplin#2922]
I am getting the following SQL error 

> Error: SQLITE_ERROR: near "Amy03": syntax error: DELETE FROM resources WHERE id IN (""Amy03-notes.jpg"")

Upon enabling encryption, but possibly related to the first sync happening thereafter. The error first shows in a popup window inside the encryption settings, and later on in the Status messages.

Checking the note in the database.sql, I can see nothing wrong with it: it exists, all referenced attachments exists as well (mid-sized jpeg files). As far as I can tell the issue is with the double quotation around the name; running the same command with single "quoting" seems to fix it.

## Environment

Joplin version: 1.0.195 (prod, linux)
Platform: Linux
OS specifics: Ubuntu 18.04.4 LTS, node 11.6.0

## Steps to reproduce

1. Enable encryption
2. Set master key
3. Popup message appears

## Logfile

[joplin.log](https://github.com/laurent22/joplin/files/4391999/joplin.log)

**Comment by joesfer** [comment]
Update: I am consistently getting this error now when I try syncing. Even after I disabled encryption.

**Comment by RedDocMD** [comment]
@joesfer, what is the sync target you are using, if you have no problem to share that. I am running this version on Ubuntu 19.10. With no sync target there is no problem.

**Comment by joesfer** [comment]
Hi, this was syncing to a directory (Dropboxes in this case, but not through the Dropbox API).

It seems as if its now trying to apply some pending updates after the failed encryption attempt, and that's resulting in an invalid query. I can try fishing the data out from the sql database if you point me to what you are after?

Thanks

**Comment by RedDocMD** [comment]
Well yes, getting to have a look at the database.sqlite file would be useful. Especially why is the DELETE command being run. I don't see any reason to delete any resource while encrypting.

**Comment by joesfer** [comment]
while I'm afraid I could not share the database per-se, I did a little bit more digging using sqlite directly (to avoid any further mess with the joplin client):

I found the note and resources ids from the note in question mentioned in the error, and found several entries in the sync_items table for the same resources. One of them was clearly stale because it pointed to sync_target=7 (dropbox), rather than 2 (file system) as the following ones did.

So I went ahead and assumed the entries were staled and perhaps the failed attempts to synchronisation were coming from this. Given that all my other clients were in sync before I attempted encryption, I went ahead and deleted all the entries from the sync_items table, closed and reopened joplin.

Sync is going ahead now, the notes look file (no data lost AFAICT) and the application seems to have happily updated all the 17k remote items. I did however get _new_ errors this time, which shed more light on the matter:

> These items will remain on the device but will not be uploaded to the sync target. In order to find these items, either search for the title or the ID (which is displayed in brackets above).
 "Amy03-notes.jpg" ("Amy03-notes.jpg") could not be uploaded: File not found: /home/jose/.config/joplin-desktop/resources/"Amy03-notes.jpg".jpeg
"Amy01-notes.jpg" ("Amy01-notes.jpg") could not be uploaded: File not found: /home/jose/.config/joplin-desktop/resources/"Amy01-notes.jpg".jpeg

The files indeed no longer exist in the mentioned location, as they were sucessfully imported into joplin long time ago. Yet they seem to be referenced somewhere. The note in question which _should_ contain said files does so (in the form of resource ids displaying file).

So I am _guessing_ here that encryption might have been a red herring, and simply have triggered a re-visit of all the (possibly stale?) entries in the sync_items table, causing the error?

The desktop client now looks fine, so I'll now try to begin encryption again and see if either succeeds or I can reproduce the problem again.

**Comment by RedDocMD** [comment]
One thing I must mention that in the `$HOME/.config/joplin-desktop/resources` directory, resources ar e saved as `<id>.<file_extension>`. In your case the file `Amy01-notes.jpg` has been saved not only by its name, but also with extra quotes. It is as though the filename right from the first time you added it as a resource had those extra quotes. Are you sure that you had not accidentally saved that file with the quotes? (It is possible in Linux to save files with names having special characters, like the following command will do the trick: `mv  Amy01-notes.jpg '"Amy01-notes.jpg"'`)

**Comment by joesfer** [comment]
Hi @RedDocMD yeah that's a good point.

I think I found the core issue: somehow, and I'm not sure how this happened, I had said entries in the resources database, where the id field was in the form of the quoted filename. This explains the SQL query format error (meant to include an id, not a string). 

I went ahead and purged the invalid entries, since they don't correspond to any valid resource, re-synced, encrypted, and everything seems good now!

Thanks for your time on this. I think we can call this one data corruption on my side, and eventually not related to encryption.
